### PR TITLE
allow zerosized tensors

### LINKED ISSUE #1498: Can't create tensor with empty buffer in metal, gpu
```
# ====== Works ======

In [21]: Tensor([], device="TORCH").realize()
Out[21]: <Tensor <LB (0,) dtypes.float op=buffer<0, dtypes.float> st=ShapeTracker(shape=(0,), views=[View(shape=(0,), strides=(1,), offset=0, mas
k=None, contiguous=True, shape_strides=((0, 1),))])> on TORCH with grad None>

In [28]: Tensor([], device="LLVM").realize()
Out[28]: <Tensor <LB (0,) dtypes.float op=buffer<0, dtypes.float> st=ShapeTracker(shape=(0,), views=[View(shape=(0,), strides=(1,), offset=0, mask=None, contiguous=True, shape_strides=((0, 1),))])> on LLVM with grad None>


# ====== Doesn't Work ======

In [25]: Tensor([], device="GPU").realize()
LogicError: create_buffer failed: INVALID_BUFFER_SIZE

In [26]: Tensor([], device="METAL").realize()
AttributeError: 'NoneType' object has no attribute 'contents'
```

Not sure if this works on CUDA since I don't have nvidiagpu

### PR DESCRIPTION
**Pull request by nimlgen**
Closes #1498

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/runtime/lib.py                    1               1
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

**Comment by ZenginU** [comment]
Hello, even with lib.py change still throws error.

**Comment by nimlgen** [comment]
> Hello, even with lib.py change still throws error. `ValueError: could not broadcast input array from shape (0,) into shape (1,)`

Thanks, fixed as well.

File: tinygrad/runtime/ops_gpu.py [hunk 1]
```diff
@@ -48,9 +48,11 @@ def synchronize(self):
 class CLBuffer(RawBufferCopyInOut, RawBufferTransfer):
   def __init__(self, size, dtype, device='0'): super().__init__(size, dtype, allocator=CL.cl_allocator, **{'device': device})
   def _copyin(self, x:np.ndarray):
+    if self.size == 0: return
```
Comment (wozeparrot) [inline tinygrad/runtime/ops_gpu.py:51]: should these checks be in `RawBufferCopyInOut` instead? are they needed for the other backends?

Comment (nimlgen) [reply tinygrad/runtime/ops_gpu.py]: Yeah, that was the only backend to get segfault with size 0. But moved to Buffers, agree this is a bit cleaner.

**Pull request merged**